compiler/accelerators/snax_gemmx.py

```diff
@@ -0,0 +1,189 @@
+from collections.abc import Sequence
+
+from xdsl.dialects import arith, builtin, llvm
+from xdsl.dialects.builtin import i8, i32
+from xdsl.ir import Operation, SSAValue
+
+import compiler.dialects.kernel as kernel
+from compiler.accelerators.dispatching import DispatchTemplate, SupportedKernel
+from compiler.accelerators.snax import SNAXAccelerator, SNAXStreamer
+from compiler.accelerators.streamers import (
+    Streamer,
+    StreamerConfiguration,
+    StreamerType,
+)
+from compiler.dialects import accfg, snax_stream
+
+default_streamer = StreamerConfiguration(
+    [
+        Streamer(  # A
+            StreamerType.Reader,
+            temporal_dims=("n", "n", "n", "n", "n", "n"),
+            spatial_dims=("n", "i", "n"),
+        ),
+        Streamer(  # B
+            StreamerType.Reader,
+            temporal_dims=("n", "n", "n"),
+            spatial_dims=("n", "n", "i"),
+        ),
+        Streamer(  # D8
+            StreamerType.Writer,
+            temporal_dims=("n", "n", "n"),
+            spatial_dims=("i", "n", "n"),
+        ),
+        Streamer(  # C
+            StreamerType.Reader,
+            temporal_dims=("n", "n", "n"),
+            spatial_dims=("i", "n", "n"),
+        ),
+        Streamer(  # D32
+            StreamerType.Writer,
+            temporal_dims=("n", "n", "n"),
+            spatial_dims=("i", "n", "n"),
+        ),
+    ]
+)
+
+
+class SNAXGEMMXAccelerator(SNAXAccelerator, SNAXStreamer, DispatchTemplate):
+    """
+    Accelerator Interface class for SNAX GEMMX accelerator
+    """
+
+    name = "snax_gemmx"
+
+    supported_kernels = (
+        SupportedKernel(kernel.QMacOp, (i8, i8, i32, i32, i32)),
+        SupportedKernel(kernel.RescaleOp, (i32, i8)),
+    )
+
+    def __init__(
+        self, streamer_config: StreamerConfiguration = default_streamer
+    ) -> None:
+        super().__init__(streamer_config)
+
+        self.fields = (
+            *self.streamer_setup_fields,
+            "K",
+            "N",
+            "M",
+            "subtractions",
+            "csr0",
+            "csr1",
+            "csr2",
+            "temporal_loop_bound",
+            "bypassSIMD",
+        )
+
+        self.launch_fields = (*self.streamer_launch_fields, "launch_gemmx")
+
+    def generate_acc_op(self) -> accfg.AcceleratorOp:
+        """
+        Return a SNAX GEMMX accelerator op
+        """
+
+        # base address:
+        base_addr = 0x3C0
+
+        addr_next, streamer_setup = self.get_streamer_setup_dict(base_addr)
+        addr_next, streamer_launch = self.get_streamer_launch_dict(addr_next)
+
+        op = accfg.AcceleratorOp(
+            self.name,
+            {
+                **streamer_setup,
+                "K": addr_next + 0,
+                "N": addr_next + 1,
+                "M": addr_next + 2,
+                "subtractions": addr_next + 3,
+                "csr0": addr_next + 4,
+                "csr1": addr_next + 5,
+                "csr2": addr_next + 6,
```

Comment:
what are these for?

Reply:
https://github.com/KULeuven-MICAS/snax_cluster/blob/a182ddc0c50ef7b64d14c7c8222dcba0cb009f14/target/snitch_cluster/sw/snax/gemmx/src/snax-gemmx-lib.c#L23
bin packed csr registers to set a variety of values. although the naming is very bad i kept it to stay consistent with the naming in hw. I"ll add some comments explaining them

Reply:
:+1: :frowning_face: :roll_eyes:  :upside_down_face: :+1: